src/transformers/processing_utils.py

```diff
@@ -993,6 +990,59 @@ def apply_chat_template(
         )
 
 
+def _check_reversed_images_text_for_vlms(images, text):
+    """
+    For backward compatibility: reverse the order of `images` and `text` inputs if they are swapped.
+    This method should only be called for processors where `images` and `text` have been swapped for uniformization purposes.
+    Note that this method assumes that two `None` inputs are valid inputs. If this is not the case, it should be handled
+    in the processor's `__call__` method before calling this method.
+    """
+
+    def _is_valid_text_input_for_processor(t):
+        if isinstance(t, str):
+            # Strings are fine
+            return True
+        elif isinstance(t, (list, tuple)):
+            # List are fine as long as they are...
+            if len(t) == 0:
+                # ... not empty
+                return False
+            elif isinstance(t[0], (str, int)):
+                # ... list of strings or int (for encoded inputs)
+                return True
+            elif isinstance(t[0], (list, tuple)):
+                # ... list of list of strings or int (for encoded inputs)
+                return isinstance(t[0][0], (str, int))
+            else:
+                return False
+        else:
+            return False
+
+    def _is_valid_or_convertible(input, validator, converter):
+        is_valid = validator(input) or input is None
+        is_convertible = converter(input) if not is_valid else False
+        return is_valid, is_convertible
+
+    images_is_valid, images_is_text = _is_valid_or_convertible(
+        images, valid_images, _is_valid_text_input_for_processor
+    )
+    text_is_valid, text_is_images = _is_valid_or_convertible(text, _is_valid_text_input_for_processor, valid_images)
```

Comment:
In code, we should avoid patterns when we do `def a_and_b` - as the functions are no longer atomic and therefore cannot be well composed. Instead, this should be split up 

The converter code we don't even need wrapped up in a function

```suggestion
    def _is_valid(input, validator):
        return validator(input) or input is None

    images_is_valid = _is_valid(images, valid_images)
    images_is_text = _is_valid_text_input_for_processor(images) if not images_is_valid else False
    
    text_is_valid = _is_valid(text, _is_valid_text_input_for_processor)
    text_is_images = valid_images(text) if not text_is_valid else False
```